crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -393,6 +398,7 @@ fn simplify_call(func: ValueId, arguments: &[ValueId], dfg: &mut DataFlowGraph)
             let limb_count = constant_args[2].to_u128() as u32;
             SimplifiedTo(constant_to_radix(endian, field, radix, limb_count, dfg))
         }
+        // TODO: Can simplify here too!
         Intrinsic::BlackBox(_) | Intrinsic::Println | Intrinsic::Sort => None,
```

Comment:
For BlackBox I think simplifying each function there would require essentially re-implementing each one in SSA entirely. This is possible but perhaps a bit cumbersome and opens us up to inconsistencies if its implementation differs from acir-gen.

For Println we cannot simplify since it would lead to printing each item out of order as each println in the program has its arguments become known constants at different times (or not).

For Sort I think we can add a simplification. It should be simple enough to check if its argument is a constant array and if so do `array.sort()`.

Reply:
We can reuse the solvers from ACVM here so the only new code we need is to read arguments and create new constants. We'll need to be careful if we change the interface of a black box function as we'll need to update how we read the arguments but I don't see this being a common thing.

see #1981